torchvision/prototype/datasets/_builtin/caltech.py

```diff
@@ -0,0 +1,209 @@
+    Mapper,
+    TarArchiveReader,
+    Shuffler,
+    Filter,
+)
+
+from torchdata.datapipes.iter import KeyZipper
+from torchvision.prototype.datasets.utils import (
+    Dataset,
+    DatasetConfig,
+    DatasetInfo,
+    HttpResource,
+    OnlineResource,
+)
+from torchvision.prototype.datasets.utils._internal import create_categories_file, INFINITE_BUFFER_SIZE, read_mat
+
+HERE = pathlib.Path(__file__).parent
+
+
+class Caltech101(Dataset):
+    @property
+    def info(self) -> DatasetInfo:
+        return DatasetInfo(
+            "caltech101",
+            categories=HERE / "caltech101.categories",
+            homepage="http://www.vision.caltech.edu/Image_Datasets/Caltech101",
+        )
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+        images = HttpResource(
+            "http://www.vision.caltech.edu/Image_Datasets/Caltech101/101_ObjectCategories.tar.gz",
+            sha256="af6ece2f339791ca20f855943d8b55dd60892c0a25105fcd631ee3d6430f9926",
+        )
+        anns = HttpResource(
+            "http://www.vision.caltech.edu/Image_Datasets/Caltech101/Annotations.tar",
+            sha256="1717f4e10aa837b05956e3f4c94456527b143eec0d95e935028b30aff40663d8",
+        )
+        return [images, anns]
+
+    _IMAGES_NAME_PATTERN = re.compile(r"image_(?P<id>\d+)[.]jpg")
+    _ANNS_NAME_PATTERN = re.compile(r"annotation_(?P<id>\d+)[.]mat")
+    _ANNS_CATEGORY_MAP = {
+        "Faces_2": "Faces",
+        "Faces_3": "Faces_easy",
+        "Motorbikes_16": "Motorbikes",
+        "Airplanes_Side_2": "airplanes",
+    }
+
+    def _is_not_background_image(self, data: Tuple[str, Any]) -> bool:
+        path = pathlib.Path(data[0])
+        return path.parent.name != "BACKGROUND_Google"
+
+    def _is_ann(self, data: Tuple[str, Any]) -> bool:
+        path = pathlib.Path(data[0])
+        return bool(self._ANNS_NAME_PATTERN.match(path.name))
+
+    def _images_key_fn(self, data: Tuple[str, Any]) -> Tuple[str, str]:
+        path = pathlib.Path(data[0])
+
+        category = path.parent.name
+        id = self._IMAGES_NAME_PATTERN.match(path.name).group("id")
+
+        return category, id
+
+    def _anns_key_fn(self, data: Tuple[str, Any]) -> Tuple[str, str]:
+        path = pathlib.Path(data[0])
+
+        category = path.parent.name
+        if category in self._ANNS_CATEGORY_MAP:
+            category = self._ANNS_CATEGORY_MAP[category]
+
+        id = self._ANNS_NAME_PATTERN.match(path.name).group("id")
+
+        return category, id
+
+    def _collate_and_decode_sample(
+        self, data, *, decoder: Optional[Callable[[io.IOBase], torch.Tensor]]
+    ) -> Dict[str, Any]:
+        key, image_data, ann_data = data
+        category, _ = key
+        image_path, image_buffer = image_data
+        ann_path, ann_buffer = ann_data
+
+        label = self.info.categories.index(category)
+
+        image = decoder(image_buffer) if decoder else image_buffer
+
+        ann = read_mat(ann_buffer)
```

Comment:
Does `ann` contain anything else or just `box_coord` and `obj_contour`?

Reply:
Nope.

Reply:
Just `box_coord` and `obj_contour`?